pagopa/io-app
# chore: [IOBP-341] Update IDPayPaymentCodeInputScreen to use new DS components

## Short description
updated said screen to use new DS components
<img src="https://github.com/pagopa/io-app/assets/60693085/f63ac10d-d077-49a6-b5b1-eac4d7d2df19" width=250/>

## List of changes proposed in this pull request
- added new `TextInput` component to screen
- screen no longer uses deprecated NB-based `FooterWithButtons`

## How to test
with the idpay flag active, navigate to an IdPay discount initiative, tap "authorize payment", navigate to the last of the three items at the bottom of the screen and make sure the page follows the design.
